controllers/movieController.js

```diff
@@ -0,0 +1,96 @@
+const Movie = require("../models/Movies");
+
+// List all movies from MongoDB database
+exports.movies_list = function(req, res, next) {
+  req.app.locals.db
+    .collection("movies")
+    .find({})
+    .toArray((err, result) => {
+      if (err) {
+        res.status(400).send({ error: err });
+      }
+      if (result === undefined || result.length === 0) {
+        res.status(400).send({ error: "No movies in database" });
+      } else {
+        res.status(200).send(result);
+      }
+    });
+};
+
+// Select a movie by id from database
+exports.movie_details = function(req, res, next) {
+  req.app.locals.db.collection("movies").findOne(
+    {
+      _id: req.params.id
+    },
+    (err, result) => {
+      if (err) {
+        res.status(400).send({ error: err });
+      }
+      if (result === undefined) {
+        res.status(400).send({ error: "No movie matching that id was found" });
+      } else {
+        res.status(200).send(result);
+      }
+    }
+  );
+};
+
+// Add a new movie to database
+exports.movie_add = function(req, res, next) {
+  const newMovie = new Movie(
+    req.body.title,
+    req.body.year,
+    req.body.author,
+    req.body.body
+  );
+  req.app.locals.db.collection("movies").insertOne(
```

Comment:
I'm not a big fan of putting the callback of a function underneath, I didn't get it at first.
I'd write it as:
```
... .insertOne({ newMovie }, (err, res) => {
  // code
}
```
Try to use the state space of the screen to convey for information in each line and separate concerns vertically. But as I said, it's just a matter of preference.

Reply:
Solved!